### ISSUE TITLE
parameter_order check does not work default hash is added in parameter

### ISSUE THREAD

**Issue by rmestrum** [rodjek/puppet-lint#544]
The check parameter_order has a false positive when a parameter of a class is provided with a default hash that contains for example a top scope variable

```
$var1 = 'test'

class test (
  $entries = {
    '200 xxx' => {
      param1      => $::var1,
      param2     => 'value2',
      param3     => 'value3',
    }
  }
) {
 notice $entries
}

class {'test':
}
```

When running puppet-lint 2.0.2 the output is as followed:

```
$ puppet-lint --no-documentation-check --no-2sp_soft_tabs-check --no-autoloader_layout-check test.pp 
WARNING: optional parameter listed before required parameter on line 6
```

**Comment by bastelfreak** [comment]
Hi, we hit the same today:
```puppet
class jira::facts(
  $ensure        = 'present',
  $port          = $jira::tomcat_port,
  $contextpath   = $jira::contextpath,
  $json_packages = $jira::params::json_packages,
  $uri           = $jira::tomcat_address ? { 
    undef   => '127.0.0.1',
    default => $jira::tomcat_address,
  },  
) inherits jira::params {
...
}
```

```
manifests/facts.pp:24:parameter_order:WARNING:optional parameter listed before required parameter
```
